Dockerfile

```diff
@@ -28,15 +30,15 @@ ARG UI_PUBLIC_URL=/
 ENV UI_PUBLIC_URL=${UI_PUBLIC_URL}
 
 ARG VERSION=unknown-branch
+ENV VERSION=${VERSION}
 ARG TARGETARCH
+ENV TARGETARCH=${TARGETARCH}
 # hadolint ignore=SC2155
-RUN export BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%S+00:00(UTC)'); \
-    export REACT_APP_BUILD_DATE=${BUILD_DATE}; \
-    export COMMIT_SHA="-=<GitHub-CI-commit-sha-placeholder>=-"; \
-    export REACT_APP_BUILD_COMMIT_SHA=${COMMIT_SHA}; \
-    export VERSION="${VERSION}-${TARGETARCH}"; \
-    export REACT_APP_BUILD_VERSION=${VERSION}; \
-    npm run build -- --base="${UI_PUBLIC_URL}"
+RUN export REACT_APP_BUILD_DATE="$(date -u +'%Y-%m-%dT%H:%M:%S+00:00(UTC)')"; \
+    export REACT_APP_BUILD_COMMIT_SHA="-=<GitHub-CI-commit-sha-placeholder>=-"; \
+    export REACT_APP_BUILD_VERSION="${VERSION}-${TARGETARCH}"; \
+    export REACT_APP_API_ENDPOINT="${REACT_APP_API_ENDPOINT}"; \
+    npm run build"$( [ "$NODE_ENV" = "production" ] || echo ":dev" )" -- --base="${UI_PUBLIC_URL}"
```

Comment:
How about `npm run build$( [ "$NODE_ENV" != "production" ] && echo ":dev" ) -- ...`? The condition matching the branch is more clear to me.

Also the quote is not necessary as it's already quoted by `$(...)`.